subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ApplicationPluginIntegrationTest.groovy

```diff
@@ -497,4 +498,20 @@ rootProject.name = 'sample'
         File scriptOutputDir = file('build/scripts')
         new File(scriptOutputDir, filename)
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/1923")
+    def "not up-to-date if classpath changes"() {
+        given:
+        succeeds("startScripts")
+
+        when:
+        buildFile << """
```

Comment:
could you explain why this triggers the issue in a comment here? e.g., something like...

```
// Set a version so the jar is created with a different name
// This should cause us to regenerate the script 
```